src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs

```diff
@@ -239,31 +239,81 @@ private IAzureContext MigrateSecretToKeyStore(IAzureContext context, AzKeyStore
+                            account.ExtendedProperties.SetProperty(AzureAccount.Property.ServicePrincipalSecret, secret);
+                        }
+                        catch
+                        {
+                        }
+                    }
+                    if (account?.Type == AzureAccount.AccountType.ServicePrincipal && !account.IsPropertySet(AzureAccount.Property.CertificatePassword))
+                    {
+                        try
+                        {
+                            var secret = keystore.GetSecureString(new ServicePrincipalKey(AzureAccount.Property.CertificatePassword, account.Id, account.GetTenants().First())).ConvertToString();
+                            account.ExtendedProperties.SetProperty(AzureAccount.Property.CertificatePassword, secret);
+                        }
+                        catch
+                        {
+                        }
+                    }
+                }
+
+            }
+            return ret;
         }
 
+        private void LoadImpl(string contents)
+        {
+        }
 
         /// <summary>
-        /// Creates new instance of AzureRMProfile.
+        /// Creates new instance of AzureRMProfile with environment table initialized.
         /// </summary>
-        public AzureRmProfile()
+        public AzureRmProfile(bool initializeEnvironmentTable = true):this()
         {
-            EnvironmentTable = new ConcurrentDictionary<string, IAzureEnvironment>(StringComparer.CurrentCultureIgnoreCase);
-
-            // Adding predefined environments
-            foreach (var env in AzureEnvironment.PublicEnvironments)
+            if (initializeEnvironmentTable)
             {
-                EnvironmentTable[env.Key] = env.Value;
+                EnvironmentTable = new ConcurrentDictionary<string, IAzureEnvironment>(StringComparer.CurrentCultureIgnoreCase);
+
+                // Adding predefined environments
+                foreach (var env in AzureEnvironment.PublicEnvironments)
+                {
+                    EnvironmentTable[env.Key] = env.Value;
+                }
             }
         }
 
+        /// <summary>
+        /// Creates new instance of AzureRMProfile. used only in Json conversion.
+        /// </summary>
+        private AzureRmProfile()
+        {
+        }
+
         /// <summary>
         /// Initializes a new instance of AzureRMProfile and loads its content from specified path.
         /// </summary>
         /// <param name="path">The location of profile file on disk.</param>
         /// <param name="shouldRefreshContextsFromCache"></param>
-        public AzureRmProfile(string path, bool shouldRefreshContextsFromCache = true) : this()
+        public AzureRmProfile(string path, bool shouldRefreshContextsFromCache = true) : this(true)
```

Comment:
to be sure, is this(true) expected?

Reply:
> to be sure, is this(true) expected?

Define another constructor to avoid confusion and please review it.